packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,155 @@
+      participantLeftNoticeString: locale.participantLeftNoticeString,
+      twoParticipantLeftNoticeString: locale.twoParticipantLeftNoticeString,
+      threeParticipantLeftNoticeString: locale.threeParticipantLeftNoticeString,
+      unnamedParticipantString: locale.unnamedParticipantString,
+      manyParticipantsJoined: locale.manyParticipantsJoined,
+      manyParticipantsLeft: locale.manyParticipantsLeft,
+      manyUnnamedParticipantsJoined: locale.manyUnnamedParticipantsJoined,
+      manyUnnamedParticipantsLeft: locale.manyUnnamedParticipantsLeft
+    };
+  }, [locale]);
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  const currentParticipants = useSelector(getRemoteParticipantsConnectedSelector);
+  const [previousParticipants, setPreviousParticipants] = useState<RemoteParticipantState[]>(currentParticipants);
+
+  const resetAnnoucement = (string: string): void => {
+    // Is this really needed?
+    // React does not promise that we will see a DOM update with '' value.
+    // If we're getting announcer to announce the same name twice in sequence, it's probably keyed off of the
+    // actual DOM node, which will update even if we don't reest to '' first.
+    setAnnouncerString('');
+    setAnnouncerString(string);
+  };
+
+  useMemo(
+    () => {
+      const whoJoined = currentParticipants.filter((p) => !previousParticipants.includes(p));
+      const whoLeft = previousParticipants.filter((p) => !currentParticipants.includes(p));
+      if (whoJoined.length > whoLeft.length) {
+        resetAnnoucement(createAnnouncmentString('joined', whoJoined, strings));
+      }
+      if (whoLeft.length > whoJoined.length) {
+        resetAnnoucement(createAnnouncmentString('left', whoLeft, strings));
+      }
+      // Update cached value at the end.
+      setPreviousParticipants(currentParticipants);
+    },
+    // previousParticipants caches the value of `currenParticipants`. We _don't_ want this
+    // hook to run for when `previousParticipants` is updated.
+    // If we did, the second run would always clear out the value of `whoJoined` etc.
+    [currentParticipants, previousParticipants, strings]
+  );
+  return announcerString;
+};
+
+/**
+ * Generates the announcement string for when a participant joins or leaves a call.
+ */
+export const createAnnouncmentString = (
+  direction: 'joined' | 'left',
+  participants: RemoteParticipantState[],
+  strings: ParticipantChangedAnnouncmentStrings
+): string => {
+  /**
+   * Filter participants into two arrays to put all the unnamed participants at the back of the
+   * names being announced.
+   */
+  const unnamedParticipants = participants.filter((p) => p.displayName === undefined);
+  const namedParicipants = participants.filter((p) => p.displayName);
+  const sortedParticipants = namedParicipants.concat(unnamedParticipants);
+
+  /**
+   * if there are only unnamed participants present in the array announce a special unnamed participants
```

Comment:
Moving comment from #2502, what if there are two unnamed participants and one named participant?

Reply:
Can we get confirmation from a11y team on how to handle multiple participants without set display names

Reply:
A11y team is happy with these strings as long as the user has the opportunity to see the list of participants somewhere else in the app.